### ISSUE TITLE
Base colorscheme switching between light/dark does not work

### ISSUE THREAD

**Issue by dawnonmercury** [shaunsingh/nyoom.nvim#34]
Currently, there is a light and dark mode of the colorscheme in highlights.fnl. However, switching between bg=light/dark fails to properly switch between them.

With `(set! background :dark)` it (as expected) automatically uses the dark theme on opening nvim. Manually setting `:set bg=light` breaks the colorscheme completely.

Commenting out `(set! background :dark)` has the exact same result. When bg is automatically set to dark by the terminal, the colorscheme works correctly but breaks on manual `:set bg=light`. Opening nvim when bg is automatically set to light by the terminal breaks the colorscheme.

With `(set! background :light)` in highlihgts.fnl, the same happens in converse. The lights scheme loads correctly, but manually using `:set ng=dark` breaks the colorscheme completely.

As it stands, the only way to really switch between light and dark modes would be to edit highlights.fnl before opening nvim. There is a possibility that this can only be fixed using an external plugin, such as lush.nvim, which would very much be not ideal.

**Comment by shaunsingh** [comment]
When you run `:set bg=light`, it changes the value of `background` but doesn't reload `highlights.fnl` (as the color scheme isn't a vimscript color scheme e.g. something you would load with `colorscheme -`, its just a list of highlights I manually require). `highlights.fnl`

So if you comment out the line `(set! background :dark)` in `highlights.fnl`, run `:set bg=light`, then run `:source %` on the `highlights.fnl` buffer, it should set it correctly. 

In other words, `:set bg=light` is doing whats expected in your issue, and to have vim properly reload it upon changing bg between light/dark I would need to define it as a colorscheme under the `colors/` dir. I'm thinking of turning the highlights file into its own base16 themeing module, which you can then use with telescope to change between themes.

**Comment by shofel** [comment]
@shaunsingh you're right this behaviour is easily explained by the difference between nyoom.nvim and plain neovim.

:thinking: But as a second point of `Design` section in readme we've got:
>Close to metal. There's less between you and vanilla neovim by design. That's **less to grok and less to work around** when you tinker.

### Colors could behave like in a plain neovim

So I think it would be nice if colorschemes work as in plain neovim. To make it so we should:

1. extract (set! background :dark) to :core.options
2. make a thin wrapper, which re-loads `highlights.fnl`, and place [in `colors/` folder](https://neovim.io/doc/user/syntax.html#:~:text=:colo[rscheme]%20{name})
3. load the highlights not with `require`, but with `vim.cmd "colorscheme nyoom-carbon"`

### Is it better at all?

Pros:
- for the users, colors work exactly as in plain vim

cons:
- another level of indirection in the nyoom code introduced by using vim command

Other words, this step would absorb inconsistency faced by the users, and encapsulate it in the nyoom's internals. As for me, this is definitely a good thing

**Comment by shaunsingh** [comment]
Small update: The issue on `hotpot.nvim` above discusses a build system that allows you to build fennel files into rtp dirs. This issue is fairly easy to solve (just have to require the `highlights.fnl` file from the `colors/` dir, but I wanted to keep the config all fennel (for fun). After the build system is implemented, we can build `core/highlights.fnl` into `colors/carbondark.lua`, making It a "normal" neovim/vim color scheme

**Comment by shaunsingh** [comment]
Using the build api (thank you hotpot!), neovim watches for changes on `fnl/carbon/carbon.fnl` and compiles it to `colors/carbon.lua`, where you can source it using `colorscheme carbon`. This hasn't fixed the issue with switching between light/dark but thats an easy fix 

> another level of indirection in the nyoom code introduced by using vim command

luckily now we have the `nvim_cmd` api. in combination with a nice [macro](https://github.com/shaunsingh/nyoom.nvim/blob/401aa1077fda24c0a4873edfa609c7e12422851e/fnl/macros/highlight-macros.fnl#L3) solves the vim command and the indirection issues

**Comment by shaunsingh** [comment]
Oddly if a color scheme is loaded using `colorscheme ...` then it should be reloaded upon changing the value of `background`, but it isn't. I'm not sure if thats an `nvim_cmd` issue or a neovim-specific issue. 

For now I just manually reload the colorscheme. You can use hydra (keymap `SPC o b`) to toggle between light/dark

https://user-images.githubusercontent.com/71196912/181173571-ce20235a-a8fc-4e18-b7cc-5ac9931b46e8.mov

Is this a good solution? feel free to let me know if you'd like something different

**Comment by dawnonmercury** [comment]
Sorry for the late reply, but yeah this works perfectly. Down the line it would be nice to figure out how to get the colorscheme to automatically detect BG=light/dark, but this works perfectly.
